### PR TITLE
Textual Rules -> Rule Name Uniqueness

### PR DESCRIPTION
**Pull request by ConfusedTA**
Add information on requirement for uniqueness of rule names.

Signed-off-by: Garry Mitchell garrymitchell@my105e.com (github: ConfusedTA)

### CODE REVIEW

**Comment by ThomDietrich** [comment]
Hey @ConfusedTA ,
that's a good addition. I'd suggest to claim that the rule name should always be in quotes.

Also as these sentences below the syntax address all three parts of the syntax I think it would be better to turn the whole thing into a itemized description. A quick example (please don't use):

- Rule name: ....
- Condition: ...
- Logic: ...

**Comment by ConfusedTA** [comment]
Thanks for the feedback @ThomDietrich ,

I have amended the commit, as well as changing the example rule to reflect this suggestion.

File: configuration/rules-dsl.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ Each rule has the following syntax:
 ```java
 rule "rule name"
```
Comment (ThomDietrich) [inline configuration/rules-dsl.md]:
```

```java
rule "<RULE_NAME>"
```

File: configuration/rules-dsl.md [hunk 2]
```diff
@@ -101,8 +101,9 @@ then
 end
 ```
 
-A rule can have any number of trigger conditions, but must at least have one.
-The `SCRIPT_BLOCK` contains the code that should be executed, when a trigger condition is met, see the [script](#scripts) section for details on its syntax.
+- Each rule must have a unique name within your openHAB installation, and the name should be wrapped in quotes.
+- A rule can have any number of trigger conditions, but must at least have one.
+- The `SCRIPT_BLOCK` contains the code that should be executed, when a trigger condition is met, see the [script](#scripts) section for details on its syntax.
```
Comment (ThomDietrich) [inline configuration/rules-dsl.md]: I was thinking about something similar to:

- `<RULE_NAME>` - Each rule must have a unique name (given in quotes). It is recommended to choose a speaking meaningful description as rule name.
- `<TRIGGER_CONDITION>` - A rule is executed in reaction to at least one or multiple trigger conditions. Please see below for different possible triggers.
- `<SCRIPT_BLOCK>` - Contains the logic that should be executed when a trigger condition is met. See the [script](#scripts) section for details on its code syntax.

**Comment by ConfusedTA** [comment]
I was considering putting it as a list too - updated!

**Comment by ThomDietrich** [comment]
Great work @ConfusedTA ! Thanks for the help. Feel free to add further PRs with additions. The rules articles can benefit from that...

From my point of view the PR is fine, I'll wait for @bgilmer77 to do the final spelling and grammar check and merge 👍

File: configuration/rules-dsl.md [hunk 3]
```diff
@@ -90,19 +90,17 @@ The **Rules** section contains a list of rules.
...
 
 ```java
-rule "rule name"
+rule "<RULE_NAME>"
 when
-    <TRIGGER CONDITION1> or
-    <TRIGGER_CONDITION2> or
-    <TRIGGER_CONDITION3>
-    ...
+    <TRIGGER_CONDITION>
```
Comment (ThomDietrich) [inline configuration/rules-dsl.md]: Ah, this change... How about
```java
    <TRIGGER_CONDITION> [or <TRIGGER_CONDITION2>]
```

File: configuration/rules-dsl.md [hunk 4]
```diff
@@ -90,19 +90,17 @@ The **Rules** section contains a list of rules.
...
 then
     <SCRIPT_BLOCK>
 end
 ```
 
-A rule can have any number of trigger conditions, but must at least have one.
-The `SCRIPT_BLOCK` contains the code that should be executed, when a trigger condition is met, see the [script](#scripts) section for details on its syntax.
+- `<RULE_NAME>` - Each rule must have a unique name (given within quotes). It is recommended to choose a meaningful description as rule name.
```
Comment (bgilmer77) [inline configuration/rules-dsl.md]: Per the comment from @ThomDietrich, consider, "It is recommended that you choose a name that has meaning when spoken."

File: configuration/rules-dsl.md [hunk 5]
```diff
@@ -90,19 +90,17 @@ The **Rules** section contains a list of rules.
...
+- `<TRIGGER_CONDITION>` - The triggering event upon which the rule logic is executed. At least one condition has to be given, multiple conditions are separated by the keyword `or`. Please see below for different possible triggers.
```
Comment (bgilmer77) [inline configuration/rules-dsl.md]: Based upon @ThomDietrich's comments, please consider the following.  "<TRIGGER_CONDITION>- A rule is executed in reaction to one or more trigger conditions.  Please see below for possible triggers."

Comment (ThomDietrich) [reply configuration/rules-dsl.md]: I'd leave the first sentence, or at least it's notion.
It is there to clearly state WHAT trigger condition is.
After that sentence you may describe as much as you want or all I care ;)

Comment (bgilmer77) [reply configuration/rules-dsl.md]: Agree completely.

**Comment by ConfusedTA** [comment]
Great to get a bit more feedback, have implemented changes based upon your input.

**Comment by Confectrician** [comment]
2 Maintainer reviews and one approve. No need to wait longer. 😃 

Thanks from my side too @ConfusedTA

**Pull request merged**